python/cudf/cudf/_lib/nvtx/colors.py

```diff
@@ -0,0 +1,32 @@
+import functools
+
+_NVTX_COLORS = {
+    "green": 0x008000,
+    "blue": 0x0000FF,
+    "yellow": 0xFFFF00,
+    "purple": 0x800080,
+    "cyan": 0x00FFFF,
+    "red": 0xFF0000,
+    "white": 0xFFFFFF,
+    "darkgreen": 0x006400,
+    "orange": 0xFFA500,
+}
+
+
+@functools.lru_cache()
+def color_to_hex(color):
+    """
+    Convert color to ARGB hex value.
+    """
+    if color in _NVTX_COLORS:
+        return _NVTX_COLORS[color]
+    try:
+        import matplotlib.colors
+    except ImportError as e:
+        raise TypeError(
+            f"Invalid color {color}. Please install matplotlib "
+            "for additional colors support"
+        ) from e
+    rgba = matplotlib.colors.to_rgba(color)
+    argb = (rgba[-1], rgba[0], rgba[1], rgba[2])
+    return int(matplotlib.colors.to_hex(argb, keep_alpha=True)[1:], 16)
```

Comment:
This looks kinda expensive, could you do a quick `timeit` on this to see how much it costs?

Reply:
It's about `100ns` (lru_cached). Disabling the cache, it's about `10us`.

Reply:
We could maintain our own color->hex mapping or just vendor in matplotlib's. The caveat is that matplotlib supports things like provided a tuple of RGB/RGBA values.

Reply:
Also we'll always need to post-process RGBA->ARGB as I think the former is more common in the PyData ecosystem, while the latter is what NVTX needs. There's probably a better way than I'm doing?

Reply:
`100ns` with the lru_cache is pretty good, thanks!